docs/migration-to-airlift.md

```diff
@@ -0,0 +1,177 @@
+at port 9081, and the Admin service at port 9082. The new version only
+listens to one single port.
+
+### Merge of request service and application service (:9080 and :9081)
+Requests to the Trino Gateway (Web UI and RESTful API) and requests that need to
+be forwarded to Trino are both served by the same port. The destination is
+determined by the HTTP URI in the request.
+
+### Removal of Admin service (:9082)
+The admin console page provided by Dropwizard (port 9082) has been removed.
+JMX can be viewed at `/v1/jmx`. There are no plans to support other functions on
+the Dropwizard Admin page.
+
+### Path change for WebUI
+The root path for Trino Gateway WebUI has been changed from `/` to
+`/trino-gateway`. By default, access to `/` will be redirected to `/trino-gateway`.
+This behavior can be changed by adding `/` to `extraWhitelistPaths`, which
+will route `/` to Trino.
+
+## Change of configuration
+### Change of service ports
+`requestRouter`, `server`, `applicationConnectors`, and `adminConnectors`
+have been replaced by `serverConfig`.
+
+Old config:
+```yaml
+requestRouter:
+    port: 8080
+    name: trinoRouter
+    historySize: 1000
+    requestBufferSize: 8192
+server:
+    applicationConnectors:
+        - type: http
+          port: 8090
+          useForwardedHeaders: true
+    adminConnectors:
+        - type: http
+          port: 8091
+          useForwardedHeaders: true
+```
+
+New config:
+```yaml
+serverConfig:
+    node.environment: test
+    http-server.http.port: 8443
+    proxy.http-client.request-buffer-size: 8kB
+```
+
+### TLS Config
+`ssl`, `keystorePath`, and `keystorePass` have been replaced by
+`http-server.https.*`. For more details, see: [Security](security.md)
+
+Old config:
+```yaml
+requestRouter:
+  ssl: true
+  port: 8080
+  name: trinoRouter
+  historySize: 1000
+  keystorePath: <path>
+  keystorePass: <password>
+
+server:
+  applicationConnectors:
+    - type: https
+      port: 8090
+      keyStorePath: <path>
+      keyStorePassword: <password>
+      useForwardedHeaders: true
+  adminConnectors:
+    - type: https
+      port: 8091
+      keyStorePath: <path>
+      keyStorePassword: <password>
+      useForwardedHeaders: true
+```
+
+New config:
+```yaml
+serverConfig:
+    http-server.http.enabled: false
+    http-server.https.enabled: true
+    http-server.https.port: 8080
+    http-server.https.keystore.path: <path>
+    http-server.https.keystore.key: <password>
+```
+
+### Logging
+Old config:
+
+* Set logging to external
+```yaml
+logging:
+    type: external
+```
+* Also specify the path to the `log.properties` file using the `log.levels-file`
+  JVM options, such as `-Dlog.levels-file=etc/log.properties`.
+
+New config:
+
+* Specify the path to the `log.properties` file in config
+```yaml
+serverConfig:
+    log.levels-file: gateway-ha/etc/log.properties
+```
+*  The `log.levels-file` JVM option is no longer supported.
+
+### Startup arguments
+The first arg `server` has been removed.
```

Comment:
Doesnt that just use the server mode for the JVM and we still probably want that?

Reply:
Nope, that's the dropwizard thing.
https://www.dropwizard.io/en/stable/getting-started.html#running-your-application

I think `-server` is almost the default option in modern JDK and we don't need to set it.